### PR TITLE
Add eigh() to torchx

### PR DESCRIPTION
**Pull request by howard0su**

### CODE REVIEW

**Comment by howard0su** [comment]
Not ready yet.

File: torchx/lib/torchx/backend.ex [hunk 1]
```diff
@@ -642,6 +642,13 @@ defmodule Torchx.Backend do
     |> to_nx(out)
   end
 
+  @impl true
+  def eigh({eigenvals, eigenvecs}, tensor, _opts) do
+    {q, r} = Torchx.eigh(from_nx(tensor))
```
Comment (josevalim) [inline torchx/lib/torchx/backend.ex:647]:
```suggestion
  def eigh({eigenvals, %T{type: type} = eigenvecs}, tensor, _opts) do
    {q, r} =
      tensor
      |> from_nx()
      |> Torchx.to_type(to_torchx_type(type))
      |> Torchx.eigh()
```

Comment (polvalente) [reply torchx/lib/torchx/backend.ex:647]: it should be {eigenvecs, eigenvals} instead of {q, r}

Comment (polvalente) [reply torchx/lib/torchx/backend.ex:647]: And I would rename the output holder by adding the _out suffix

**Review by josevalim — APPROVED** [review]

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**